### ISSUE TITLE
In my saved file, Seebeck coefficient of n-type and p-type are the same values. how could I solve this problem?

### ISSUE THREAD

**Issue by wavechayanon** [K4ys4r/BoltzTraP_Tools#3]
[SvNreSe.zip](https://github.com/K4ys4r/BoltzTraP_Tools/files/4770631/SvNreSe.zip)

**Comment by K4ys4r** [comment]
You saved the data twice !
You saved S=f(N) at 300, 600 and 900K ! but you did it twice for each temperature.

**Comment by wavechayanon** [comment]
Yes, but I save it for n-type and p-type in each temperature. So my saved file should have 2 types in each column of the same temperature.

**Comment by K4ys4r** [comment]
Can you put all the step that you do and the trace file.

**Comment by wavechayanon** [comment]
This is my bash script

#!/bin/bash
<<:
This Script Excute the Script_S_scaling.py and outputs the plot of
Seebeck at 300,500 and 800 K as a function of Energy.
:

python Script_S_scaling.py <<EOF
n
Trace
Hf.trace
0.00
T
y
N
S
300
electron
yes
300
hole
yes
600
electron
yes
600
hole
yes
900
electron
yes
900
hole
no
save
SvNSC.data
no
EOF

**Comment by wavechayanon** [comment]
my trace file
[Hf.zip](https://github.com/K4ys4r/BoltzTraP_Tools/files/4814880/Hf.zip)

**Comment by K4ys4r** [comment]
Actually in the saved file the program save all the value (n and p Type); because it is complicated to save two different type for reason there are not the same date row. For exemple, if we have for n-type data 300 rows and for p-type 200 rows it is complicated to save them into one file because there x values is different.
You can post-processing the data.

**Comment by wavechayanon** [comment]
Could you explain the post-processing to find the seebeck coefficient vs doping level for n-type and p-type from my file?
I'm a beginner in this field. 
Thank you in advance.

**Comment by K4ys4r** [comment]
Here is a script to handle your data file.

```python
import pylab as p 
import numpy as np 
# Loading data 
data = np.loadtxt("SvNreSe.data") 
  
# Get the data index for n & p Type 
indxp = np.where(data[:,0]>0)  
indxn = np.where(data[:,0]<0) 

# Plot data with log scale for N 
p.semilogx(data[indxp,0][0],data[indxp,1][0],"r",label="P-Type") 
p.semilogx(abs(data[indxn,0][0]),data[indxn,1][0],"b",label="N-Type") 
p.legend(loc="best")   
# Save data to files 
np.savetxt("pType300K.data",np.c_[data[indxp,0][0],data[indxp,1][0]])
np.savetxt("nType300K.data",np.c_[data[indxn,0][0],data[indxn,1][0]]) 
```

Hope that helps you.
Hilal

**Comment by immortal-Research** [comment]
Hello Dear K4ys4r can you please guide me on how to install BoltzTraP tool. It is given in the guide that it is written in python 2, but when I am running the command python2 setup.py install it is giving me following error - 
Traceback (most recent call last):
  File "setup.py", line 1, in <module>
    from setuptools import setup
ImportError: No module named setuptools

**Comment by K4ys4r** [comment]
Hi,
The error message told you that on your system you don’t have the setup tools module! Then you should install it.
https://pypi.org/project/setuptools/
Regards
Hilal;
